### DISCUSSION TITLE
writing a single byte data in modbus

### DISCUSSION THREAD

**Discussion by sunilkumar105** [pymodbus-dev/pymodbus#1303] [Q&A] [answered]
Hii !
hope this question finds you well.
I am working on a project where i am writing some data to PLC after doing some computations in RASPBERRY PI 4. 
for example the data i am writing is 9888989, so what i am doing is i am packing this data into 4 bytes and after that writing the data one byte to every register in modbus. starting from address 5000
i am using .write_register function (code 0x06) which is for writing single register
but the problem is after writing this data to plc, when i am reading it back in plc ,
there data appears like after a gap of a byte. 
for example the byte1, byte2, bytt3 & byte4 are getting written at address 5000,5001,5002,5003 but when reading from the plc it is appearing like byte1 at address 0x0, byte2 at 0x2, byte3 at address 0x4, byte4 at address 0x6 and the address 0x1,0x3,0x5&0x7 are 0 always.

 function code 0x06 and 0x10 are used to read and write analog holding register respectively and the length of analog holding registers are 2 bytes.
and length of input register are also 2 bytes
is it possible to write byte by byte at each address
if so please suggests
Thankyou

**Comment by janiversen** [comment, marked as answer]
Write_register automatically converts the value you send to big endian (depending on your CPU) for each register. However you need to check if a register swap (words) is needed.

In other words, the byte swapping is done automatically because it is within 1 register, but word swapping is responsibility of the app.

No it is not possible to write bytes, a register is defined as 2 bytes. so each address is 16bit wide, not 8bit.

**Comment by janiversen** [comment]
Please read the modbus specification https://modbus.org/docs/Modbus_Application_Protocol_V1_1b3.pdf
